android/lib/architecture-test/src/test/java/cz/quanti/rocketapp/android/lib/architecturetest/model/ArchitectureLayer.kt

```diff
@@ -0,0 +1,114 @@
+package cz.quanti.rocketapp.android.lib.architecturetest.model
+
+import com.lemonappdev.konsist.api.architecture.Layer
+
+enum class ArchitectureLayer(val layerName: String) {
+    /**
+     * Can depend on:
+     * presentation, data
+     *
+     * Contains:
+     * Screens, Activity, Application, Delegates, Jetpack Compose components, Services, ...
+     * Implementation of Dao or Api or other data provider interfaces (if it is required by the tool that is used)
+     */
+    System("system"),
+
+    /**
+     * Can depend on
+     * domain, model
+     *
+     * Contains
+     * ViewModels, View State data classes, mappers from domain models
+     */
+    Presentation("presentation"),
+
+    /**
+     * Can depend on
+     * model
+     *
+     * Contains
+     * UseCase interfaces + implementation
+     * Repository interfaces
+     */
+    Domain("domain"),
+
+    /**
+     * Cannot depend on any other layer
+     *
+     * Contains
+     * domain level models (data classes)
+     */
+    Model("model"),
+
+    /**
+     * Can depend on
+     * domain, model
+     *
+     * Contains
+     * Repository implementations
+     * data source interfaces (for Dao, Api, ...)
+     * data level entities and mappers to model entities
+     */
+    Data("data"),
+
+    /**
+     * Should not depend on other layers
+     *
+     * Contains
+     * general utils/helper/extension functions that extend the functionality of the language or other project wide tool
+     */
+    Infrastructure("infrastructure"),
+
+    /**
+     * Can depend on
+     * all layers
+     *
+     * Contains
+     * koin module and definitions
+     */
+    Di("di"),
+}
+
+fun ArchitectureLayer.toKonsistLayer(): Layer {
+    return Layer(name, "${PackagePatterns.ROOT_PACKAGE}..$layerName..")
+}
+
+fun ArchitectureLayer.dependencies(): List<ArchitectureLayer> = when (this) {
```

Comment:
udelal jsem extension na dependencies, prijde mi to citelnejsi nez primo v testu. Nelze dat dependecy primo do enum, protoze je to cyklicka zavislost..